pallets/gear/src/internal.rs

```diff
@@ -315,25 +325,45 @@ where
 
         // Deadline of the task.
         //
-        // NOTE: make sure to work around it, while doing db migrations,
-        // changing `ReserveFor` value.
+        // NOTE: the `ReserveFor` value may have changed due to storage migration thereby
+        // leading to a mismatch between the amount due and the amount deposited upfront.
         let deadline = hold_interval
             .finish
             .saturating_add(CostsPerBlockOf::<T>::reserve_for());
 
-        // The block number, which was the last payed for hold.
+        // The block number, which was the last paid for hold.
         //
-        // Outdated tasks can store for free, but this case is under
-        // control of correct `ReserveFor` constant set.
-        let payed_till = current.min(deadline);
+        // Outdated tasks can end up being store for free - this case has to be controlled by
+        // a correct selection of the `ReserveFor` constant.
+        let paid_until = current.min(deadline);
 
         // Holding duration.
-        let duration: u64 = payed_till
+        let duration: u64 = paid_until
             .saturating_sub(hold_interval.start)
             .unique_saturated_into();
 
+        // Cost per block based on the storage used for holding (identifiable by `lock_id`)
+        let cost = match lock_id {
+            LockIdentifier::Mailbox => CostsPerBlockOf::<T>::mailbox(),
+            LockIdentifier::Waitlist => CostsPerBlockOf::<T>::waitlist(),
+            LockIdentifier::Reservation => CostsPerBlockOf::<T>::reservation(),
+            _ => CostsPerBlockOf::<T>::dispatch_stash(),
```

Comment:
Bad practice for future changes
```suggestion
            LockIdentifier::DispatchStash => CostsPerBlockOf::<T>::dispatch_stash(),
```